cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -1107,7 +1107,7 @@ private ArtifactCell(Id.Artifact artifactId) {
 
     ArtifactData(Location location, ArtifactMeta meta) {
       this.locationURI = null;
-      this.locationPath = Locations.getRelativePath(location);
+      this.locationPath = Locations.getRelativePath(location.getLocationFactory(), location.toURI());
```

Comment:
you didn't need to change all of these, you could have just added the new getRelativePath method and had the existing one call it.